bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyBaseHandler.java

```diff
@@ -302,43 +303,38 @@ public void handleCommand(ChannelUID channelUID, Command command) {
             if (update) {
                 requestUpdates(1, false);
             }
-        } catch (NullPointerException | IOException e) {
-            if (authorizationFailed(e.getMessage())) {
+        } catch (ShellyApiException e) {
+            ShellyApiResult res = e.getApiResult();
+            if (isAuthorizationFailed(res)) {
                 return;
             }
-            if (e.getMessage().contains(APIERR_NOT_CALIBRATED)) {
-                logger.warn("Device is not calibrated, use Shelly App to perform initial roller calibration.");
+            if (res.isNotCalibrtated()) {
+                logger.warn("{}:{}", thingName, messages.get("roller.calibrating"));
             } else {
-                logger.warn("{} ERROR: Unable to process command for channel {}: {} ({})\nStack Trace: {}", thingName,
-                        channelUID.toString(), e.getMessage(), e.getClass(), e.getStackTrace());
+                logger.info("{}: {}{}", thingName, messages.get("command.failed", command, channelUID), e.toString());
             }
-        } finally {
-            lockUpdates = false;
+        } catch (IllegalArgumentException e) {
+            logger.debug("{}: {}{}", thingName, messages.get("command.failed", command, channelUID), e.toString());
         }
     }
 
     /**
      * Update device status and channels
      */
-    @SuppressWarnings("null")
-    protected void updateStatus() {
+    protected void refreshStatus() {
         try {
             boolean updated = false;
 
             skipUpdate++;
-            if (lockUpdates) {
-                logger.trace("{}: Update locked, try on next cycle", thingName);
-                return;
-            }
-
-            if ((skipUpdate % refreshCount == 0) && (profile != null)
+            if ((skipUpdate % refreshCount == 0) && (profile.isInitialized())
                     && (getThing().getStatus() == ThingStatus.ONLINE)) {
```

Comment:
Please make a local variable out of `getThing().getStatus()` so you can reuse it.

Reply:
changed